packages/core/src/modules/problem-reports/errors/ProblemReportError.ts

```diff
@@ -0,0 +1,20 @@
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { ProblemReportMessage } from '../messages/ProblemReportMessage'
+
+export interface ProblemReportErrorOptions {
+  problemCode: string
+}
+
+export class ProblemReportError extends AriesFrameworkError {
+  public problemReport: ProblemReportMessage
+
+  public constructor(message: string, { problemCode }: ProblemReportErrorOptions) {
+    super(message, undefined)
+    this.problemReport = new ProblemReportMessage({
+      description: {
+        en: message,
+        code: problemCode,
+      },
```

Comment:
Ah I see you used the problem report message in error approach here. I think we can also leverage that in the connection / credential / proofs messages